pkg/app/app.go

```diff
@@ -170,3 +154,26 @@ func (a *App) Start() {
 		os.Exit(1)
 	}
 }
+
+func initClients(log *logging.Logger, ec clients.EtcdConfig) {
+	// Designed to panic early if we cannot construct required clients.
+	// this likely means we're in an unrecoverable configuration or enviornment.
+	// Best we can do is alert the operator as early as possible.
+	//
+	// Deliberately forcing the injection of deps here instead of running as a
+	// method on the app. Forces developers at authorship time to think about
+	// dependencies / make sure things are ready.
+	log.Notice("Initializing clients...")
+	log.Debug("Trying to connect to etcd")
+	serv, clust, err := clients.GetEtcdVersion(ec)
+	if err != nil {
+		log.Error("Failed to connect to Etcd\n")
+		log.Error(err.Error())
```

Comment:
Honestly I way prefer to see things on new lines so I'm not reading shitty wrapped stack traces or errors. Will update to Errorf.